crates/sui-core/src/authority_active.rs

```diff
@@ -176,33 +181,185 @@ impl<A> ActiveAuthority<A>
 where
     A: AuthorityAPI + Send + Sync + 'static + Clone,
 {
-    pub async fn spawn_all_active_processes(self) {
-        self.spawn_active_processes(true, true).await
+    pub async fn spawn_all_active_processes(self, receiver: LifecycleTaskHandler) {
+        self.spawn_active_processes(receiver, true, true).await
     }
 
     /// Spawn all active tasks.
-    pub async fn spawn_active_processes(self, gossip: bool, checkpoint: bool) {
-        // Spawn a task to take care of gossip
+    pub async fn spawn_active_processes(
+        self,
+        receiver: LifecycleTaskHandler,
+        gossip: bool,
+        checkpoint: bool,
+    ) {
+        let gossip_receiver = receiver.clone();
         let gossip_locals = self.clone();
-        let _gossip_join = tokio::task::spawn(async move {
-            if gossip {
-                gossip_process(&gossip_locals, 4).await;
+        let gossip_join = gossip_receiver.spawn("Gossip".to_string(), move || {
+            let inner_locals = gossip_locals.clone();
+            async move {
+                if gossip {
+                    gossip_process(&inner_locals, 4).await;
+                }
+                Ok(())
             }
         });
 
-        // Spawn task to take care of checkpointing
         let checkpoint_locals = self; // .clone();
-        let _checkpoint_join = tokio::task::spawn(async move {
-            if checkpoint {
-                checkpoint_process(&checkpoint_locals, &CheckpointProcessControl::default()).await;
+        let checkpoint_join = receiver.spawn("Gossip".to_string(), move || {
+            let inner_locals = checkpoint_locals.clone();
+            async move {
+                if checkpoint {
+                    checkpoint_process(&inner_locals, &CheckpointProcessControl::default()).await;
+                }
+                Ok(())
             }
         });
 
-        if let Err(err) = _gossip_join.await {
-            error!("Join gossip task end error: {:?}", err);
-        }
-        if let Err(err) = _checkpoint_join.await {
-            error!("Join checkpoint task end error: {:?}", err);
+        join!(gossip_join, checkpoint_join);
+    }
+}
+
+/*
+    The active authority start-update-shutdown facilities.
+
+    The active authority runs in a separate task, and manages the tasks of separate active
+    processes, with one sub-task per active process. These subtasks internally listen to a
+    watch channel on which others may send a start, shutdown, or update signal. When this
+    is received the tasks are restarted, or shut down.
+
+*/
```

Comment:
small nit: its generally more accepted to use single line comments for comments like this, eg `// `

Reply:
Now changed!